File: app/src/main/java/com/nasahapps/materialdesigntoolbox/example/ui/components/StepperListFragment.java
package com.nasahapps.materialdesigntoolbox.example.ui.components;

import android.graphics.Color;
import android.os.Bundle;
import android.support.v4.app.ListFragment;
import android.support.v4.content.ContextCompat;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import com.nasahapps.materialdesigntoolbox.components.StepperProgressLayout;
import com.nasahapps.materialdesigntoolbox.example.R;
import com.nasahapps.materialdesigntoolbox.example.ui.main.MainActivity;

/**
 * Created by Hakeem on 4/13/16.
 */
public class StepperListFragment extends ListFragment {

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        setListAdapter(new ArrayAdapter<>(getActivity(), android.R.layout.simple_list_item_1,
                new String[]{
                        "Stepper Horizontal Layout",
                        "Stepper Progress w/Text",
                        "Stepper Progress w/Dots",
                        "Stepper Progress w/Bar"
                }));
    }

    @Override
    public void onResume() {
        super.onResume();
        ((MainActivity) getActivity()).setToolbarColor(ContextCompat.getColor(getContext(),
                R.color.nh_indigo_500));
        ((MainActivity) getActivity()).setStatusBarColor(ContextCompat.getColor(getContext(),
                R.color.nh_indigo_700));
        ((MainActivity) getActivity()).setToolbarTitleTextColor(Color.WHITE);
        ((MainActivity) getActivity()).setToolbarTitle("Steppers");
    }

    @Override
    public void onListItemClick(ListView l, View v, int position, long id) {
        super.onListItemClick(l, v, position, id);

        switch (position) {
            case 0:
                ((MainActivity) getActivity()).startFragment(new StepperHorizonalFragment());
                break;
            case 1:
                ((MainActivity) getActivity()).startFragment(StepperProgressFragment.newInstance(StepperProgressLayout.TYPE_TEXT));
                break;
            case 2:
                ((MainActivity) getActivity()).startFragment(StepperProgressFragment.newInstance(StepperProgressLayout.TYPE_DOTS));
                break;
            case 3:
                ((MainActivity) getActivity()).startFragment(StepperProgressFragment.newInstance(StepperProgressLayout.TYPE_BAR));
                break;
        }
    }
}

File: app/src/main/java/com/nasahapps/materialdesigntoolbox/example/ui/components/StepperListFragment.kt
package com.nasahapps.materialdesigntoolbox.example.ui.components

import android.graphics.Color
import android.os.Bundle
import android.support.v4.app.ListFragment
import android.support.v4.content.ContextCompat
import android.view.View
import android.widget.ArrayAdapter
import android.widget.ListView

import com.nasahapps.materialdesigntoolbox.components.StepperProgressLayout
import com.nasahapps.materialdesigntoolbox.example.R
import com.nasahapps.materialdesigntoolbox.example.ui.main.MainActivity

/**
 * Created by Hakeem on 4/13/16.
 */
class StepperListFragment : ListFragment() {

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        listAdapter = ArrayAdapter(activity, android.R.layout.simple_list_item_1,
                arrayOf("Stepper Horizontal Layout", "Stepper Progress w/Text", "Stepper Progress w/Dots", "Stepper Progress w/Bar"))
    }

    override fun onResume() {
        super.onResume()
        (activity as? MainActivity)?.let {
            it.setToolbarColor(ContextCompat.getColor(context,
                    R.color.mdt_indigo_500))
            it.setStatusBarColor(ContextCompat.getColor(context,
                    R.color.mdt_indigo_700))
            it.setToolbarTitleTextColor(Color.WHITE)
            it.setToolbarTitle("Steppers")
        }
    }

    override fun onListItemClick(l: ListView?, v: View?, position: Int, id: Long) {
        super.onListItemClick(l, v, position, id)

        (activity as? MainActivity)?.let {
            when (position) {
                0 -> it.startFragment(StepperHorizonalFragment())
                1 -> it.startFragment(StepperProgressFragment.newInstance(StepperProgressLayout.TYPE_TEXT))
                2 -> it.startFragment(StepperProgressFragment.newInstance(StepperProgressLayout.TYPE_DOTS))
                3 -> it.startFragment(StepperProgressFragment.newInstance(StepperProgressLayout.TYPE_BAR))
            }
        }
    }
}
